proc_CPMG.py

```diff
@@ -87,43 +68,63 @@ def print_fun(x, f, accepted):
             T=1000.
             )
     zeroorder_rad, firstorder = sol.x
+    angle = (zeroorder_rad)/pi*180 
+    #applying phase shift
     phshift = exp(-1j*2*pi*f_axis*(firstorder*1e-6))
     phshift *= exp(-1j*2*pi*zeroorder_rad)
     s *= phshift
-    print("RELATIVE PHASE SHIFT WAS {:0.1f}\\us and {:0.1f}$^\circ$".format(
-            firstorder,angle(zeroorder_rad)/pi*180))
+    logger.info(strm("RELATIVE PHASE SHIFT WAS {:0.1f}\\us and {:0.1f}$^\circ$".format(firstorder,angle)))
+    fl.show();quit()
     if s['nEchoes',0].data[:].sum().real < 0:
         s *= -1
-    print(ndshape(s))
+    logger.info(strm(ndshape(s)))
     fl.next('after phased - real ft')
     fl.image(s.real)
     fl.next('after phased - imag ft')
     fl.image(s.imag)
-    #data = s['t2':0]
-    data = s['t2':(-200,200)].sum('t2')
-    #data = s['t2':(0,200)].sum('t2')
+
+    #summing along t2 axis and plotting initial decay
+    data = s['t2':(-100,150)].sum('t2')
     fl.next('Echo decay')
-    x = tE_axis
+    x = s.getaxis('nEchoes')
     ydata = data.data.real
     ydata /= max(ydata)
     fl.plot(x,ydata,'-o', alpha=0.7, label='%s'%label_str, human_units=False)
+    print(s.dimlabels)
+    #fl.show();quit()
+    print(ndshape(s))
+    print("Beginning T2 curve")
+    s_sliced = s['nEchoes',(0,60)]['t2',(-50,100)]
+    s = fitdata(s_sliced)
+    M0,T2,tE = sympy.symbols("M_0 T_2 t_E", real=True)
+    s.functional_form = M0*sympy.exp(-tE/T2)
+    print("Functional form", s.functional_form)
+    print("Function string",s.function_string)
+    s.fit_coeff = r_[-1,1,1]
+    fl.next('T2 test')
+    fl.plot(s,'o',label=f.name())
+    print("symbolic variable:",s.symbolic_vars)
+    fl.show();quit()
+    
+    
+    fl.show();quit()
```

Comment:
don't quit in the middle of the scripts -- if you want to set up a conditional option, then do that, but if the stuff below is outdated code, delete it (I think this is related to the previous comment)

Reply:
done